File: clients/node/MastodonService.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MoostodonService = void 0;
const mastodonClient_1 = require("./sdk/mastodonClient");
const kiota_http_fetchlibrary_1 = require("@microsoft/kiota-http-fetchlibrary");
// Create Wrapper service around the MoostodonClient
class MoostodonService {
    constructor(baseurl) {
        const adapter = new kiota_http_fetchlibrary_1.FetchRequestAdapter(new AnonymousAuthProvider());
        this._client = new mastodonClient_1.MastodonClient(adapter);
        adapter.baseUrl = baseurl;
        console.log("initialized client with baseurl: " + adapter.baseUrl);
    }
    get client() {
        return this._client;
    }
    // Get a  user's account details from the Moostodon Client
    getAccount(accountId) {
        return __awaiter(this, void 0, void 0, function* () {
            console.log("getting account");
            return yield this._client.api.v1.accountsById(accountId).get();
        });
    }
    // Get the mastodon public timeline
    getPublicTimeline() {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this._client.api.v1.timelines.public.get();
        });
    }
}
exports.MoostodonService = MoostodonService;
class AnonymousAuthProvider {
    authenticateRequest(request, additionalAuthenticationContext) {
        return __awaiter(this, void 0, void 0, function* () {
            // do nothing
        });
    }
}

File: clients/python/mastodon_service.py
import asyncio

from typing import List
from kiota_http.httpx_request_adapter import HttpxRequestAdapter
from kiota_abstractions.authentication.anonymous_authentication_provider import AnonymousAuthenticationProvider
from sdk.models.status import Status
from sdk.models.account import Account
from sdk.mastodon_client import MastodonClient
from sdk.api.v2.search.search_request_builder import SearchRequestBuilder


class MastodonService():
    """
    This class is responsible for the communication with the Mastodon API using the MoostodonClient.
    """
    def __init__(self, base_url: str) -> None:
        """
        Instantiates a new MoostodonService and sets the default values.
        """
        
        provider = AnonymousAuthenticationProvider()
        request_adapter = HttpxRequestAdapter(provider)
        self.client = MastodonClient(request_adapter)
        request_adapter.base_url = base_url
    
    def get_public_statuses(self) -> List[Status]:
        """
        Gets the statuses from the Moostodon API.
        Returns:
            The list of statuses.
        """
        statuses = asyncio.run(self.client.api().v1().timelines().public().get())
        return statuses

    def get_account(self, account_id: str) -> Account:
        """
        Get account details for a user.
        """
        
        return asyncio.run(self.client.api().v1().accounts_by_id(account_id).get())

    def search_accounts(self, q: str) -> List[Account]:
        """
        Search for accounts
        """
        config = SearchRequestBuilder.SearchRequestBuilderGetRequestConfiguration()
        config.query_parameters = SearchRequestBuilder.SearchRequestBuilderGetQueryParameters()
        config.query_parameters.q = q
        config.query_parameters.type = "accounts"
        searchResult = asyncio.run(self.client.api().v2().search().get(request_configuration=config))
        return searchResult.accounts


